### PR TITLE
Pull out parties uploader into library component

### PR DESCRIPTION
**Pull request by alexmatson-da**
The changes in `dabl-react` are compiled code

### CODE REVIEW

File: ui/dabl-react/lib/DablPartiesInput.js [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+    var invalidLedger = parties.reduce(function (valid, party) { return valid || (party.ledgerId !== ledgerId); }, false);
+    // True if any token is expired
+    var invalidTokens = parties.reduce(function (valid, party) { return valid || JwtTokens_1.expiredToken(party.token); }, false);
+    if (invalidLedger) {
+        var fileLedgerId = (_a = parties.find(function (p) { return p.ledgerId !== ledgerId; })) === null || _a === void 0 ? void 0 : _a.ledgerId;
+        var errMessage = "Your parties.json file is for a different ledger! File uses ledger " + fileLedgerId + " but app is running on ledger " + ledgerId;
+        throw new InvalidPartiesError(errMessage, PartyErrors.LedgerMismatchError);
+    }
+    if (invalidTokens) {
+        throw new InvalidPartiesError('Your parties.json file contains expired tokens!', PartyErrors.ExpiredTokenError);
```
Comment (Sarahb-da) [inline ui/dabl-react/lib/DablPartiesInput.js:67]: super helpful

File: ui/dabl-react/lib/DablPartiesInput.js [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+                if (event.target && typeof event.target.result === 'string') {
+                    handleFileUpload(event.target.result);
+                }
+            };
+            if (e.target && e.target.files) {
+                reader.readAsText(e.target.files[0]);
+            }
+        } }));
+};
+//# sourceMappingURL=DablPartiesInput.js.map
```
Comment (Sarahb-da) [inline ui/dabl-react/lib/DablPartiesInput.js:137]: new line at end of file

File: ui/dabl-react/lib/DablPartiesInput.js.map [hunk 3]
```diff
@@ -0,0 +1 @@
...,SAAS,CAAC;QACd,IAAI,WAAW,EAAE;YACT,IAAA,4DAA4E,EAA1E,iBAAO,EAAE,eAAiE,CAAC;YACnF,IAAI,SAAO,EAAE;gBACX,UAAU,CAAC,SAAO,CAAC,CAAC;aACrB;YAED,IAAI,OAAK,EAAE;gBACT,QAAQ,CAAC,OAAK,CAAC,CAAC;aACjB;SACF;IACH,CAAC,EAAE,CAAC,WAAW,EAAE,QAAQ,EAAE,YAAY,CAAC,CAAC,CAAC;IAE1C,eAAK,CAAC,SAAS,CAAC;QACd,IAAI,OAAO,EAAE;YACX,MAAM,CAAC,OAAO,CAAC,CAAC;SACjB;IACH,CAAC,EAAE,CAAC,OAAO,CAAC,CAAC,CAAC;IAEd,eAAK,CAAC,SAAS,CAAC;QACd,IAAI,KAAK,EAAE;YACT,OAAO,CAAC,KAAK,CAAC,CAAC;SAChB;IACH,CAAC,EAAE,CAAC,KAAK,CAAC,CAAC,CAAC;IAEZ,IAAM,gBAAgB,GAAG,UAAC,QAAgB;QAClC,IAAA,yDAAyE,EAAvE,eAAO,EAAE,aAA8D,CAAC;QAEhF,IAAI,OAAO,EAAE;YACX,UAAU,CAAC,OAAO,CAAC,CAAC;SACrB;QAED,IAAI,KAAK,EAAE;YACT,QAAQ,CAAC,KAAK,CAAC,CAAC;SACjB;IACH,CAAC,CAAA;IAED,OAAO,CACL,yCAAO,IAAI,EAAC,MAAM,EAAC,KAAK,EAAC,EAAE,EAAC,QAAQ,EAAE,UAAA,CAAC;YACrC,IAAM,MAAM,GAAG,IAAI,UAAU,EAAE,CAAC;YAEhC,MAAM,CAAC,MAAM,GAAG,UAAS,KAAK;gBAC5B,IAAI,KAAK,CAAC,MAAM,IAAI,OAAO,KAAK,CAAC,MAAM,CAAC,MAAM,KAAK,QAAQ,EAAE;oBAC3D,gBAAgB,CAAC,KAAK,CAAC,MAAM,CAAC,MAAM,CAAC,CAAC;iBACvC;YACH,CAAC,CAAC;YAEF,IAAI,CAAC,CAAC,MAAM,IAAI,CAAC,CAAC,MAAM,CAAC,KAAK,EAAE;gBAC9B,MAAM,CAAC,UAAU,CAAC,CAAC,CAAC,MAAM,CAAC,KAAK,CAAC,CAAC,CAAC,CAAC,CAAC;aACtC;QACH,CAAC,GAAG,CACL,CAAC;AACJ,CAAC,CAAA"}
```
Comment (Sarahb-da) [inline ui/dabl-react/lib/DablPartiesInput.js.map:1]: new line at end of file

**Review by Sarahb-da — APPROVED** [review]
A few "no newline at the end of file" warnings. Otherwise looks good

**Review by alexgraham-da — APPROVED** [review]
looks good to me!

**Pull request merged**